apps/allocations/app/components/App/AllocationsHistory.js

```diff
@@ -63,30 +62,34 @@ const AllocationsHistory = ({ allocations }) => {
           </Amount>
         ]
       }}
-      //renderEntryExpansion={({ recipients, amount, token }) => {
-      //  return recipients.map((recipient, index) => {
-      //    const allocated = BigNumber(recipient.amount).div(amount)
-      //    return (
-      //      <Recipient key={index}>
-      //        <IdentityBadge
-      //          entity={recipient.address}
-      //          shorten={true}
-      //        />
-      //        <RecipientProgress>
-      //          <ProgressBar
-      //            value={allocated.toNumber()}
-      //            color={String(theme.accentEnd)}
-      //          />
-      //        </RecipientProgress>
-      //        <RecipientAmount theme={theme}>
-      //          { displayCurrency(BigNumber(recipient.amount)) } {' '}
-      //          {token} {' • '}
-      //          { allocated.times(100).dp(0).toNumber() }{'%'}
-      //        </RecipientAmount>
-      //      </Recipient>
-      //    )
-      //  })
-      //}}
+      renderEntryExpansion={({ recipients, amount, token }) => {
+        const totalSupports = recipients.reduce((total, recipient) => {
+          return total + Number(recipient.supports)
+        }, 0)
+        return recipients.map((recipient, index) => {
+          const allocated = BigNumber(recipient.supports).div(totalSupports)
+          return (
+            <Recipient key={index}>
+              <IdentityBadge
+                entity={recipient.candidateAddress}
+              />
+              <RecipientAllocation>
+                <RecipientProgress>
+                  <ProgressBar
+                    value={allocated.toNumber()}
+                    color={String(theme.accentEnd)}
+                  />
+                </RecipientProgress>
+                <RecipientAmount theme={theme}>
+                  { displayCurrency(BigNumber(amount).times(allocated)) } {' '}
```

Comment:
We try to avoid calculations in out render returns, they accumulate and become expensive eventually degrading the performance of the components, so it is better as a rule of thumb to define every logic in constants and use here calculated values instead, just take note for the future please.
You don't need to fix it right now, but this is undoubtedly tech debt we are accumulating here.

Reply:
Same question here. These are not constants; they are values that we need to calculate for each recipient.

Reply:
My idea here was to do something like this:

```js
const MyComponent = ({amount, allocated}) => {
    const myCalculatedValue = displayCurrency(BigNumber(amount).times(allocated).mul(whatever).exp(anotherThing))
    const anotherExpensiveLogic = (...) // some expensive calculations here
    return(
        <Whatever>{myCalculatedValue}</Whatever>
    )
}
```

For me, it keeps returns cleaner and also allows for easier refactors / extracting logic, the general idea is to make the components dumber and eventually move logic to hooks or even background worker, that would make our frontends so light, also it is a good practices since there are some places, likes projects app where adding logic to returns resulted in unmanageable performance problems in the past, specially when you need to render the same component multiple times.